code/controllers/subsystems.dm

```diff
@@ -0,0 +1,46 @@
+#define NEW_SS_GLOBAL(varname) if(varname != src){if(istype(varname)){Recover();qdel(varname);}varname = src;}
+
+/datum/subsystem
```

Comment:
I'm not sure about the exact reasoning, but I can tell you that there's six different subsystems that it handles: the top level alarm system, as well as atmosphere_alarm, camera_alarm, fire_alarm, motion_alarm and power_alarm, each of which are datums with specific procs.